org.osgi.test.assertj.monitoring/src/main/java/org/osgi/test/assertj/monitoring/MonitoringAssertionResult.java

```diff
@@ -0,0 +1,661 @@
+/*******************************************************************************
+ * Copyright (c) Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *******************************************************************************/
+
+package org.osgi.test.assertj.monitoring;
+
+import java.util.List;
+import java.util.Map;
+
+import org.assertj.core.api.ClassBasedNavigableListAssert;
+import org.assertj.core.api.Condition;
+import org.assertj.core.api.DurationAssert;
+import org.assertj.core.api.ListAssert;
+import org.assertj.core.api.ThrowableAssert;
+import org.osgi.framework.BundleEvent;
+import org.osgi.framework.FrameworkEvent;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.ServiceEvent;
+import org.osgi.test.assertj.bundleevent.BundleEventAssert;
+import org.osgi.test.assertj.frameworkevent.FrameworkEventAssert;
+import org.osgi.test.assertj.serviceevent.ServiceEventAssert;
+
+/**
+ * The Interface RuntimeMonitoringResultAssert.
+ */
+public interface MonitoringAssertionResult {
```

Comment:
What is the thinking behind this interface?

Reply:
Gives access to assertion methods. All assertions will be executed on the recorded events.